lk2nd/lk2nd-device.c

```diff
@@ -273,15 +274,33 @@ static void lk2nd_parse_panels(const void *fdt, int offset)
 	}
 
 	new = lkfdt_getprop_str(fdt, offset, "compatible", &new_len);
-	if (!new || new_len < 1)
-		return;
+	if (!new || new_len < 1) {
+		new = lkfdt_getprop_str(fdt, offset, "replace-compatible", &new_len);
+		if (!new || new_len < 1)
+			return;
+		replace_mode = true;
+		dprintf(INFO, "Panel compatible will be *replaced*\n");
+	}
 
-	/* Include space required for null-terminators */
-	panel->compatible_size = ++new_len + ++old_len;
+	if (!replace_mode) {
+		/* Include space required for null-terminators */
+		panel->compatible_size = ++new_len + ++old_len;
+	} else {
+		panel->compatible_size = ++new_len;
+	}
 
 	panel->compatible = malloc(panel->compatible_size);
 	ASSERT(panel->compatible);
-	panel->old_compatible = panel->compatible + new_len;
+
+	if (!replace_mode) {
+		/* Old compatible points into the middle of allocated block? */
```

Comment:
Yes, due to it being `<new compatible>\0<old compatible>\0` so the comment is out of place

Reply:
this code is not present now